### ISSUE TITLE
Why error ValueAfterTable caused?

### ISSUE THREAD

**Issue by ancheGT** [toml-rs/toml#395]
I have a json:
```json
{
    "sync": {
        "storage_type": "",
        "local_disk": {
            "data_dir_path": ""
        },
        "interval_seconds": 6
    }
}
```

if I just parse this json string,it works OK:
```rust
let r= r##"
{
    "sync": {
        "storage_type": "",
        "local_disk": {
            "data_dir_path": ""
        },
        "interval_seconds": 6
    }
}
"##;
 

match toml::to_string(&r) {
    Ok(toml) => {
        print!(">>> toml: {:?} \n", toml)
    }
    Err(e) => {
        print!(">>> toml error: {:?} \n", e)
    }
}
```

but,if I parse this json string into a struct,then convert the struct into toml string,it maybe cause an **ValueAfterTable** error.

Before convert into toml,print the struct converted from json string:

> Conf { sync: Sync { storage_type: "", local_disk: SyncLocalDisk { data_dir_path: "" } } }

It as same as the json string. 

The struct is:

```rust
#[derive(Serialize, Deserialize, Debug)]
struct SyncLocalDisk {
    data_dir_path: String,
}

#[derive(Serialize, Deserialize, Debug)]
struct Sync {
    storage_type: String,
    local_disk: SyncLocalDisk, // After comment this line or the flowlling line, there will have no error
    interval_seconds: u32, 
}

#[derive(Serialize, Deserialize, Debug)]
struct Encryption {
    master_password: String,
    entry_file_name: String,
    file_ext: String,
    enable_file_compress: bool,
}

#[derive(Serialize, Deserialize, Debug)]
struct Conf {
    sync: Sync,
}
```

Or we can create a new **Conf**,then convert it,  the result is the same :

```rust
let local_disk = SyncLocalDisk {
    data_dir_path: "path".to_string(),
};

let sync = Sync {
    storage_type: "storage".to_string(),
    local_disk: local_disk,
    interval_seconds: 8,                
};

let ccc = Conf { sync: sync };

match toml::to_string(&ccc) {
    Ok(toml) => {
        print!(">>> toml: {:?} \n", toml)
    }
    Err(e) => {
        print!(">>> toml error: {:?} \n", e)
    }
}
```

I have test with 0.5.9 and 0.5.10.

What's the reason? And  how to solve  it?

Thankyou

**Comment by epage** [comment]
FYI the json reproduction case you listed doesn't seem to be complete.  It seems to just send a string to `toml::to_string`.

I've made the serde_derive case easier to reproduce and did a comparison with `toml_edit`:
```rust
#!/usr/bin/env rust-script

//! ```cargo
//! [dependencies]
//! serde = { version = "1.0", features = ["derive"] }
//! toml = "0.5.10"
//! toml_edit = { version = "0.15.0", features = ["easy"] }
//! ```

use serde::Deserialize;
use serde::Serialize;

#[derive(Serialize, Deserialize, Debug)]
struct SyncLocalDisk {
    data_dir_path: String,
}

#[derive(Serialize, Deserialize, Debug)]
struct Sync {
    storage_type: String,
    local_disk: SyncLocalDisk, // After comment this line or the flowlling line, there will have no error
    interval_seconds: u32,
}

#[derive(Serialize, Deserialize, Debug)]
struct Encryption {
    master_password: String,
    entry_file_name: String,
    file_ext: String,
    enable_file_compress: bool,
}

#[derive(Serialize, Deserialize, Debug)]
struct Conf {
    sync: Sync,
}

fn main() {
    let local_disk = SyncLocalDisk {
        data_dir_path: "path".to_string(),
    };

    let sync = Sync {
        storage_type: "storage".to_string(),
        local_disk: local_disk,
        interval_seconds: 8,
    };

    let ccc = Conf { sync: sync };

    match toml::to_string(&ccc) {
        Ok(toml) => {
            println!(">>> toml:\n{}", toml)
        }
        Err(e) => {
            println!(">>> toml error: {:?}", e)
        }
    }

    match toml_edit::easy::to_string_pretty(&ccc) {
        Ok(toml) => {
            println!(">>> toml_edit:\n{}", toml)
        }
        Err(e) => {
            println!(">>> toml_edit error: {:?}", e)
        }
    }
}
```
```console
$  ./toml-395.rs
>>> toml error: ValueAfterTable
>>> toml_edit:
[sync]
storage_type = "storage"
interval_seconds = 8

[sync.local_disk]
data_dir_path = "path"
```

So you can workaround this by using `toml_edit` and this will be fixed after #340

**Comment by ancheGT** [comment]
@epage thanks bro
